### PR TITLE
Extend stub for "how to develop"

### PR DESCRIPTION
**Pull request by SpaceK33z**
I'm planning to write the whole "how to develop" page soon, but I would like to see first what you guys think of the structure.

I'm not even sure if explaining the three ways to develop (dev-server, dev-middleware, webpack watch) is the best way to go here. We could also explain the most common and easy use case with dev-server here, and explain the other two ways in a subpage.

I really would like some feedback on this.

### CODE REVIEW

File: content/how-to/develop.md [hunk 1]
```diff
@@ -1,12 +1,44 @@
...
-> output.pathinfo
-> hot module replacement
-
-> watch
-> caching
-> dev middleware
-> in memory compilation
-> dev server
+
+Introduction:
```
Comment (bebraw) [inline content/how-to/develop.md]: Introduction is implicit so you can drop this line.

File: content/how-to/develop.md [hunk 2]
```diff
@@ -1,12 +1,44 @@
...
+
+- Show the three ways to develop: webpack --watch, webpack-dev-server, webpack-dev-middleware and explain the differences (keep it short!). dev-server is the easiest one to get started.
+- dev-server and dev-middleware use in-memory compilation
+- Explain that IT SHOULD NOT BE USED IN PRODUCTION, NEVER.
```
Comment (bebraw) [inline content/how-to/develop.md]: Sounds good. You can use `W>` for that last bit.

File: content/how-to/develop.md [hunk 3]
```diff
@@ -1,12 +1,44 @@
...
+webpack-dev-server
+
+- show one use case
+- how to install
+- show how to use in CLI
+- show how to use API
+- refer to `devServer` configuration page
+- explain `inline` and `iframe` mode, and explain `inline` requires more config in API
+- make clear that there are some differences in CLI and API
+- after explaining live reload, link to HMR article
```
Comment (bebraw) [inline content/how-to/develop.md]: This section gets easier to document if we can simplify the API at some point. Now lots of gotchas.

File: content/how-to/develop.md [hunk 4]
```diff
@@ -1,12 +1,44 @@
...
+- show one use case
+- how to install
+- show how to use API
+- refer to `devServer` configuration page
+
+webpack watch mode
+
+- show one use case
+- show how to use in CLI
+- show how to use API
```
Comment (bebraw) [inline content/how-to/develop.md]: I might explain webpack watch mode first. I actually start from there myself in my material and then justify the need for wds.

Comment (SpaceK33z) [inline content/how-to/develop.md]: Good one. Done.

File: content/how-to/develop.md (hunk 3 above)
Comment (SpaceK33z) [inline content/how-to/develop.md]: I see HMR as an advanced use case, so I only want to explain HMR very shortly here, and then link to the HMR article which tells the user how to set it up exactly.

File: content/how-to/develop.md (hunk 1 above)
Comment (SpaceK33z) [inline content/how-to/develop.md]: Done.

File: content/how-to/develop.md (hunk 3 above)
Comment (bebraw) [inline content/how-to/develop.md]: Yeah, that can work.

**Comment by bebraw** [comment]
LGTM. Let's merge?

**Comment by SpaceK33z** [comment]
Cool. I'll follow up with the whole article somewhere this week (no promises).

**Comment by bebraw** [comment]
It's alright. Thanks.

**Pull request merged**